ateam_kenobi/src/plays/basic_122.cpp

```diff
@@ -51,63 +53,133 @@ std::array<std::optional<ateam_msgs::msg::RobotMotionCommand>, 16> Basic122::run
 
   goalie_skill_.runFrame(world, motion_commands);
 
-  assignAndRunStriker(available_robots, world, motion_commands);
+  std::vector<ateam_geometry::Point> assignment_positions;
+  assignment_positions.push_back(striker_skill_.getAssignmentPoint(world));
 
-  assignAndRunBlockers(available_robots, world, motion_commands);
+  const auto blocker_assignment_positions = blockers_skill_.getAssignmentPoints(world);
+  assignment_positions.insert(
+    assignment_positions.end(),
+    blocker_assignment_positions.begin(), blocker_assignment_positions.end());
+
+  std::vector<std::vector<int>> disallowed_ids;
+  if (world.double_touch_forbidden_id_) {
+    std::fill_n(
+      std::back_inserter(disallowed_ids), assignment_positions.size(),
+      std::vector<int>{});
+    disallowed_ids[0].push_back(*world.double_touch_forbidden_id_);
+  }
+
+  const auto assignments = play_helpers::assignRobots(
+    available_robots, assignment_positions,
+    disallowed_ids);
+
+  const auto & striker = assignments[0];
+
+  if (striker) {
+    runStriker(*striker, world, motion_commands[striker->id].emplace());
+  }
+
+  std::vector<Robot> blockers;
+  for (auto ind = 1ul; ind < assignments.size(); ++ind) {
+    if (assignments[ind]) {
+      blockers.push_back(*assignments[ind]);
+    }
+  }
+
+  runBlockers(blockers, world, motion_commands);
 
   return motion_commands;
 }
 
-void Basic122::assignAndRunStriker(
-  std::vector<Robot> & available_robots, const World & world,
-  std::array<std::optional<ateam_msgs::msg::RobotMotionCommand>,
-  16> & motion_commands)
+void Basic122::runStriker(
+  const Robot & striker_bot, const World & world,
+  ateam_msgs::msg::RobotMotionCommand & motion_command)
 {
-  std::vector<ateam_geometry::Point> assignable_positions;
-  assignable_positions.push_back(striker_skill_.getAssignmentPoint(world));
-  auto assignment_map = robot_assignment::assign(available_robots, assignable_positions);
-  if (assignment_map.empty()) {
-    return;
-  }
-  const auto robot_id = assignment_map.begin()->first;
-  auto maybe_robot = world.our_robots[robot_id];
-  if (!maybe_robot) {
+  play_info_["Striker ID"] = striker_bot.id;
+
+  const auto they_have_possession = doTheyHavePossession(world);
+
+  play_info_["Possession"] = they_have_possession ? "theirs" : "ours";
+
+  if (they_have_possession) {
+    const auto ball_to_bot_vec = striker_bot.pos - world.ball.pos;
+    const auto vel = ateam_geometry::normalize(ball_to_bot_vec) * 0.25;
+    motion_command.twist.linear.x = vel.x();
+    motion_command.twist.linear.y = vel.y();
     return;
   }
```

Comment:
I think this logic is a good potential target for next easy gameplay behaviors to improve

Reply:
Definitely agree